src/main/java/ch/njol/skript/events/EvtItem.java

```diff
@@ -74,10 +75,19 @@
 					.examples("on preparing craft of torch:")
 					.since("2.2-Fixes-V10");
 		}
-		Skript.registerEvent("Pick Up", EvtItem.class, PlayerPickupItemEvent.class, "[player] (pick[ ]up|picking up) [[of] %itemtypes%]")
+		if (hasEntityPickupItemEvent) {
+			//noinspection unchecked
```

Comment:
Does ECJ recognize this? My bad, should've known that unchecked cast is here. You can't annotate static initializer.

Reply:
I'm not entirely sure. Should I move it back to the class' suppressions?

Reply:
If this works, it is better not to. Check ECJ build log and see if the warning is emitted.

Reply:
These are the warnings that are produced, I'm not entirely sure if these are the warnings we are explicitly looking for. I'll admit I'm not the best at reading/understanding some of these warnings!
```
90. WARNING in /home/travis/build/SkriptLang/Skript/src/main/java/ch/njol/skript/events/EvtItem.java (at line 80)
698	Skript.registerEvent("Pick Up", EvtItem.class, new Class[] {PlayerPickupItemEvent.class, EntityPickupItemEvent.class}, "[(player|1¦entity)] (pick[ ]up|picking up) [[of] %itemtypes%]")
699	^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
700Type safety: Unchecked invocation registerEvent(String, Class<EvtItem>, Class[], String) of the generic method registerEvent(String, Class<E>, Class<? extends Event>[], String...) of type Skript
701----------
70291. WARNING in /home/travis/build/SkriptLang/Skript/src/main/java/ch/njol/skript/events/EvtItem.java (at line 80)
703	Skript.registerEvent("Pick Up", EvtItem.class, new Class[] {PlayerPickupItemEvent.class, EntityPickupItemEvent.class}, "[(player|1¦entity)] (pick[ ]up|picking up) [[of] %itemtypes%]")
704	                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
705Type safety: The expression of type Class[] needs unchecked conversion to conform to Class<? extends Event>[]
706----------
707----------
```